File: platform/processing-core/app/schemas/partner_legal.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class PartnerLegalProfileIn(BaseModel):
    legal_type: str
    country: str | None = None
    tax_residency: str | None = None
    tax_regime: str | None = None
    vat_applicable: bool = False
    vat_rate: float | None = None


class PartnerLegalDetailsIn(BaseModel):
    legal_name: str | None = None
    inn: str | None = None
    kpp: str | None = None
    ogrn: str | None = None
    passport: str | None = None
    bank_account: str | None = None
    bank_bic: str | None = None
    bank_name: str | None = None


class PartnerLegalDetailsOut(PartnerLegalDetailsIn):
    partner_id: str
    created_at: datetime | None = None
    updated_at: datetime | None = None


class PartnerLegalProfileOut(BaseModel):
    partner_id: str
    legal_type: str
    country: str | None = None
    tax_residency: str | None = None
    tax_regime: str | None = None
    vat_applicable: bool
    vat_rate: float | None = None
    legal_status: str
    details: PartnerLegalDetailsOut | None = None
    tax_context: dict | None = None


class PartnerLegalChecklistOut(BaseModel):
    legal_profile: bool
    legal_details: bool
    verified: bool


class PartnerLegalProfileResponse(BaseModel):
    profile: PartnerLegalProfileOut | None
    checklist: PartnerLegalChecklistOut

File: frontends/partner-portal/src/types/partnerLegal.ts
export interface PartnerLegalDetails {
  partner_id: string;
  legal_name?: string | null;
  inn?: string | null;
  kpp?: string | null;
  ogrn?: string | null;
  passport?: string | null;
  bank_account?: string | null;
  bank_bic?: string | null;
  bank_name?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
}

export type PartnerTaxContext = {
  tax_rate?: number | string | null;
};

export interface PartnerLegalProfile {
  partner_id: string;
  legal_type: string;
  country?: string | null;
  tax_residency?: string | null;
  tax_regime?: string | null;
  vat_applicable: boolean;
  vat_rate?: number | null;
  legal_status: string;
  details?: PartnerLegalDetails | null;
  tax_context?: PartnerTaxContext | null;
}

export interface PartnerLegalChecklist {
  legal_profile: boolean;
  legal_details: boolean;
  verified: boolean;
}

export interface PartnerLegalProfileResponse {
  profile?: PartnerLegalProfile | null;
  checklist: PartnerLegalChecklist;
}

export interface PartnerLegalProfileUpdate {
  legal_type: string;
  country?: string | null;
  tax_residency?: string | null;
  tax_regime?: string | null;
  vat_applicable?: boolean;
  vat_rate?: number | null;
}

export interface PartnerLegalDetailsUpdate {
  legal_name?: string | null;
  inn?: string | null;
  kpp?: string | null;
  ogrn?: string | null;
  passport?: string | null;
  bank_account?: string | null;
  bank_bic?: string | null;
  bank_name?: string | null;
}
